### ISSUE TITLE
Feature request - get process cpu/memory usage

### ISSUE THREAD

**Issue by ludmilmm** [NagiosEnterprises/ncpa#189]
Currently, you can monitor the number of processes running, for example:
```
https://ipaddress:5693/api/process/firefox.exe/count
{
  "value": {
    "returncode": 0, 
    "stdout": "OK: Process count for processes named firefox.exe was 1c | 'process_count_0'=1c;;;"
  }
}
```
I would be nice to be able to check the cpu/memory usage of a process. It seems this info is available under the "Top Processes" menu, and in the API, for example:
```
https://ipaddress:5693/api/processes?name=firefox.exe
{
  "value": {
    "processes": [
      {
        "mem_vms": 417779712, 
        "exe": "C:\\Program Files (x86)\\Mozilla Firefox\\firefox.exe", 
        "mem_rss": 475721728, 
        "name": "firefox.exe", 
        "cpu_percent": 8.9
      }
    ]
  }
}
```
Is there a way to query the cpu/memory usage of a process?

**Comment by jhomann** [comment]
I don't believe there is currently an API endpoint to do this. May be something that can be added in the future, it would definitely be useful.

**Comment by daVikes** [comment]
I would also like to get top 10/20 cpu/memory processes.   Is that a similar enhancement or possible now ?

Thanks!

**Comment by bgshacklett** [comment]
@daVikes, that sounds like a good place for a script. They can be loaded into the NCPA\Plugins directory, then called at /api/agent/plugin/scriptname/?token=securitytoken&arguments=arguments

Existing Nagios check scripts should work here, as well, so you don't need to write something custom.

**Comment by jm-welch** [comment]
With the caveat that I don't know yet how difficult it would be to implement this in the API, I'd actually argue that something that would likely be this universally useful would be a solid candidate for inclusion, so as to keep the overhead of deploying NCPA lower. This is especially true given the current issues with plugins on multiple platforms (#156 - seen on mac and windows, so far).

**Comment by lgroschen** [comment]
I think adding this and ensuring it works through the plugin together would be the best solution.  I will leave this issue open for it to be added to process in the API and use #156 as a bugfix for the plugins.  According to responses this happens on OSX and windows so I will test OSX 11 and Windows 7 to see if i can replicate and fix the plugin issue.

**Comment by jomann09** [comment]
I'm going to just point out, since I guess I lost these replies and never said anything in this conversation. that I think that way to change from just simple count checks for processes to more in-depth checks is something that will be coming in future releases. 

The default behaviors will stay as is, but the ability to do it will come ~ as for overhead, yeah ... we don't want to create much overhead but this wouldn't actually require any changes to how NCPA gets the data - only how it sends it back to the user. In fact, the most overhead for NCPA is the inclusion of Python, it's libraries, etc.

**Comment by ramachandran25** [comment]
I'm also looking for the same feature since it would be very difficult to distribute the additional plugin across many servers. So it would be great if we have an option to check the memory usage of the particular process.

**Comment by doubando** [comment]
i see the api already updated with cpu memory for each proccess but not able to find a way to retrieve this via command ./check_ncpa
